nifi-nar-bundles/nifi-ccda-bundle/nifi-ccda-processors/src/main/java/org/apache/nifi/processors/ccda/ExtractCCDAAttributes.java

```diff
@@ -0,0 +1,383 @@
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if it cannot be parse CDA and store content to FlowFile.")
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        getLogger().info("Loading packages");
+
+        relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+
+        properties = new ArrayList<>();
+        properties.add(SKIP_VALIDATION);
+        properties.add(PRETTY_PRINTING);
+
+        long start = System.currentTimeMillis();
+
+        // Load required MDHT packages
+        System.setProperty( "org.eclipse.emf.ecore.EPackage.Registry.INSTANCE",
+                "org.eclipse.emf.ecore.impl.EPackageRegistryImpl" );
+        CDAPackage.eINSTANCE.eClass();
+        HITSPPackage.eINSTANCE.eClass();
+        CCDPackage.eINSTANCE.eClass();
+        ConsolPackage.eINSTANCE.eClass();
+        IHEPackage.eINSTANCE.eClass();
+        timingStats.add(String.format("Loaded packages in %d ms", System.currentTimeMillis() - start));
+
+        // Initialize JEXL
+        jexl = new JexlBuilder().cache(1024).debug(false).silent(true).strict(false).create();
+        jexlCtx = new MapContext();
+
+        getLogger().info("Loading mappings");
+        loadMappings(); // Load CDA mappings for parser
+
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        getLogger().info("Processing CCDA");
+
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        if(processMap.isEmpty()) {
+            getLogger().error("Process Mapping is not loaded");
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final Boolean skipValidation = context.getProperty(SKIP_VALIDATION).asBoolean();
+        final Boolean prettyPrinting = context.getProperty(PRETTY_PRINTING).asBoolean();
+
+        long start = System.currentTimeMillis(), total = start;
+
+        ClinicalDocument cd = loadDocument(session.read(flowFile), skipValidation); // Load and optionally validate CDA document
+
+        start = System.currentTimeMillis();
+        getLogger().info("Processing elements");
+        Map<String, Object> elements = processElement(null, cd); // Process CDA element using mapping data
+
+        start = System.currentTimeMillis();
+        getLogger().info("Generating JSON");
+
+        // Initialize GSON with optional pretty printing
+        Gson gson = prettyPrinting ? new GsonBuilder().setPrettyPrinting().create() : new GsonBuilder().create();

```

Comment:
I'm not convinced this processor is the right place for the JSON conversion/I think that should be another processor. The pattern established with the other HL7 processor is for it to pass the source data through untouched and add attributes only. If it was a standard JSON schema of some sort I think it'd make more sense but it's just the JSON representation of the attributes and since they're added as attributes we can just do an AttributesToJSON to accomplish this. So I guess I'm saying I think we should remove this piece of functionality from the processor.